File: ClientApp/src/app/interfaces/Product/Response/optionsResponse.ts
import { MultipleImagesResponse } from "./multipleImagesResponse";

export interface OptionsResponse {
    id: string;
    title: string;
    description: string | null;
    price: number;
    color: string | null;
    isMale: boolean | null;
    condition: number;
    stock: string;
    quantityAvailable: number;
    images: MultipleImagesResponse[];
}

File: ClientApp/src/app/interfaces/Product/Response/OptionsResponse.cs
namespace Contracts.Model.Product.Response;
public class OptionsResponse
{
	public Guid Id { get; set; }
	public string Title { get; set; }
	public string? Description { get; set; }
	public double Price { get; set; }
	public string? Color { get; set; }
	public bool? IsMale { get; set; }
	public int Condition { get; set; }
	public string Stock => StockCalculate().ToString();
	public int QuantityAvailable { get; set; }
	public IEnumerable<MultipleImagesResponse> Images { get; set; }

	private StockStatusType StockCalculate()
	{
		if (QuantityAvailable == 0)
			return StockStatusType.OutOfStock;
		if (QuantityAvailable > 0 && QuantityAvailable < 10)
			return StockStatusType.LowStock;
		if (QuantityAvailable >= 10)
			return StockStatusType.InStock;
		return StockStatusType.InStock;
	}
}
